---
layout: page
title: OrthoFinder
fig_image: Workflow.png
#fig_caption: "Example figure to show the data generated by an OrthoFinder run"
---
OrthoFinder is a software program for phylogenetic orthology inference. 

<figure>
<img src="{{ site.github.url }}/assets/img/Workflow.png">
<figcaption>
    <div align="right"><b>OrthoFinder workflow</b></div>
</figcaption>
</figure>

It takes as input the proteomes for the species you want to analyse and from these it automatically:
* infers the orthogroups for your species
* infers a complete set of rooted gene trees
* infers a rooted species tree
* infers all orthology relationships between the genes using the gene trees
* infers gene duplication events and cross references them to the corresponding nodes on the gene and species trees
* provides comparative genomics statistics for your species

As well as large scale analyses, it can be used to carefully check individual ortholog relationships prior to experimental studies. 

Unlike other orthology inference software OrthoFinder uses gene trees. This means you can check every ortholog relationship in the gene tree it came from. The use of gene trees gives very high ortholog inference accuracy: 

<figure>
<img src="{{ site.github.url }}/assets/img/Accuracy.png">
<figcaption>
    <div align="right"><b>Quest for Orthologs benchmarks</b></div>
</figcaption>
</figure>

Despite OrthoFinder using a more rigorous, gene-tree based approach to ortholog inference it is incredibly fast! 

<figure>
<img src="{{ site.github.url }}/assets/img/Runtime.png">
<figcaption>
    <div align="right"><b>Runtime on sets of Fungi species using 16 cores</b></div>
</figcaption>
</figure>

You can read the OrthoFinder papers here:

[Emms, D.M., Kelly, S. OrthoFinder: phylogenetic orthology inference for comparative genomics. Genome Biol 20, 238 (2019)](https://doi.org/10.1186/s13059-019-1832-y)

[Emms, D.M., Kelly, S. OrthoFinder: solving fundamental biases in whole genome comparisons dramatically improves orthogroup inference accuracy. Genome Biol 16, 157 (2015)](https://doi.org/10.1186/s13059-015-0721-2)