---
layout: post
title: "1. Downloading and running OrthoFinder"
author: "David Emms"
categories: orthofinder_tutorials
tags: [documentation]
post_type: getting_started
#fig_image: Figure3_ResultsData.png
#fig_caption: "Example figure to show the data generated by an OrthoFinder run"
---

## Plan for this tutorial
In this tutorial we're going to download OrthoFinder and check we can run it on the Example Dataset. After doing that we'll be reading for the next tutorial where we run it on a more interesting set of species. All these steps will be done on the command line so that you can just copy and run the commands yourself. 

## Downloading and running OrthoFinder

1. Create a directory to work in. Open a terminal and run the commands:
```
mkdir ~/orthofinder_tutorial
cd ~/orthofinder_tutorial
```

2. Download the latest version of OrthoFinder
```
wget https://github.com/davidemms/OrthoFinder/releases/latest/download/OrthoFinder.tar.gz
```

3. Extract the package, and cd into the OrthoFinder directory
```
tar xzvf OrthoFinder.tar.gz
cd OrthoFinder/
```

4. Request OrthoFinder to print its help file
```
./orthofinder -h
```
This will print all the OrthoFinder command line options. 

5. Run OrthoFinder on the Example Dataset
```
./orthofinder -f ExampleData/
```

### 2. A quick look at the results

When you run OrthoFinder you should get something like this:
```
~/orthofinder_tutorial$ ./orthofinder -f ExampleData/

OrthoFinder version 2.3.6 Copyright (C) 2014 David Emms

2019-10-11 15:24:21 : Starting OrthoFinder
48 thread(s) for highly parallel tasks (BLAST searches etc.)
1 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "fastme -i /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/WorkingDirectory/SimpleTest.phy -o /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/WorkingDirectory/SimpleTest.tre" - ok

Dividing up work for BLAST for parallel processing
--------------------------------------------------
2019-10-11 15:24:21 : Creating diamond database 1 of 4
2019-10-11 15:24:21 : Creating diamond database 2 of 4
2019-10-11 15:24:21 : Creating diamond database 3 of 4
2019-10-11 15:24:21 : Creating diamond database 4 of 4

Running diamond all-versus-all
------------------------------
Using 48 thread(s)
2019-10-11 15:24:21 : This may take some time....
2019-10-11 15:24:30 : Done all-versus-all sequence search

Running OrthoFinder algorithm
-----------------------------
2019-10-11 15:24:30 : Initial processing of each species
2019-10-11 15:24:30 : Initial processing of species 0 complete
2019-10-11 15:24:30 : Initial processing of species 1 complete
2019-10-11 15:24:30 : Initial processing of species 2 complete
2019-10-11 15:24:30 : Initial processing of species 3 complete
2019-10-11 15:24:33 : Connected putative homologues
2019-10-11 15:24:33 : Written final scores for species 0 to graph file
2019-10-11 15:24:33 : Written final scores for species 1 to graph file
2019-10-11 15:24:33 : Written final scores for species 2 to graph file
2019-10-11 15:24:33 : Written final scores for species 3 to graph file
2019-10-11 15:24:33 : Ran MCL

Writing orthogroups to file
---------------------------
Orthogroups have been written to tab-delimited files:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups.tsv
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups.txt (OrthoMCL format)
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups_UnassignedGenes.tsv
Sequences for orthogroups:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroup_Sequences/

2019-10-11 15:24:34 : Done orthogroups

Analysing Orthogroups
=====================

Calculating gene distances
--------------------------
2019-10-11 15:24:36 : Done

Inferring gene and species trees
--------------------------------
2019-10-11 15:24:36 : Done 0 of 325
2019-10-11 15:24:36 : Done 100 of 325
2019-10-11 15:24:36 : Done 200 of 325

269 trees had all species present and will be used by STAG to infer the species tree


Best outgroup(s) for species tree
---------------------------------
2019-10-11 15:24:41 : Starting STRIDE
2019-10-11 15:24:41 : Done STRIDE
Observed 2 well-supported, non-terminal duplications. 2 support the best roots and 0 contradict them.
Best outgroups for species tree:
  Mycoplasma_hyopneumoniae
  Mycoplasma_agalactiae, Mycoplasma_hyopneumoniae
  Mycoplasma_agalactiae


Multiple potential species tree roots were identified, only one will be analyed.
================================================================================

Reconciling gene trees and species tree
---------------------------------------
Outgroup: Mycoplasma_hyopneumoniae
2019-10-11 15:24:41 : Starting Recon and orthologues
2019-10-11 15:24:41 : Starting OF Orthologues
2019-10-11 15:24:42 : Done 0 of 325
2019-10-11 15:24:42 : Done 100 of 325
2019-10-11 15:24:42 : Done 200 of 325
2019-10-11 15:24:42 : Done 300 of 325
2019-10-11 15:24:42 : Done OF Orthologues
2019-10-11 15:24:42 : Done Recon

Writing results files
=====================
2019-10-11 15:24:42 : Done orthologues
Orthogroups have been written to tab-delimited files:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups.tsv
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups.txt (OrthoMCL format)
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroups/Orthogroups_UnassignedGenes.tsv
Sequences for orthogroups:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthogroup_Sequences/


Gene trees:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Gene_Trees/

Rooted species tree:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Species_Tree/SpeciesTree_rooted.txt

Species-by-species orthologues directory:
   /home/emms/orthofinder_tutorial/ExampleData/OrthoFinder/Results_Oct11/Orthologues/

Orthogroup statistics:
   Statistics_PerSpecies.tsv   Statistics_Overall.tsv   Orthogroups_SpeciesOverlaps.tsv

OrthoFinder assigned 2202 genes (80.6% of total) to 604 orthogroups. Fifty percent of all genes were in orthogroups with 4 or more genes (G50 was 4) and were contained in the largest 281 orthogroups (O50 was 281). There were 269 orthogroups with all species present and 246 of these consisted entirely of single-copy genes.

CITATION:
 When publishing work that uses OrthoFinder please cite:
 Emms D.M. & Kelly S. (2015), Genome Biology 16:157

 If you use the species tree in your work then please also cite:
 Emms D.M. & Kelly S. (2017), MBE 34(12): 3267-3278
 Emms D.M. & Kelly S. (2018), bioRxiv https://doi.org/10.1101/267914
```

OrthoFinder creates a directory within the directory with your input files and puts all the results there, e.g.: `ExampleData/OrthoFinder/Results_Oct11`. This is what the results directory looks like:

<img src="{{ site.github.url }}/assets/img/results_directory.png">

If everything worked then you should have got a similar looking results directory. That's it, we're done!

In the next tutorial ([Running your own OrthoFinder analysis](/orthofinder_tutorials/020-running-your-own-orthofinder-analysis.html)) we will look at how to prepare and run our own analysis and after that there's a tutorial showing how to explore all the results. 